### DISCUSSION TITLE
Reduced flow in small (top) solid infill areas for better finish

### DISCUSSION THREAD

**Discussion by dtex91** [supermerill/SuperSlicer#2393] [Show and tell]
Dear Superslicer Team, 

for some trials i have coded a litte program which analyzes the gcode after I have sliced a model with superslicer to kind of add a little function to it. Whenever i print something (with a tuned, good EM/flow), little areas keep getting overextruded. I have found not method to reduce that in superslicer itself. Because of that, i first coded something quick and dirty to just see if my idea/method can improve my top surface print quality.

For my opinion, i can see some good improvements. So my question is, if its possible to bring this into Slicer? Or may I try to bring it in myself (im a software developer). What do you think?

How am I doing this? I have defined several iterations of path length like 2mm, 3mm, 4mm, 6mm, 8mm....and within these iterations i have reduced the amount of how much the extruder has to drive. I guess in future one can replace those iterations with a formula, but i have to calculate one soon. 
This reduced flow doesnt just happen for top solid infill. Its als crucial to use it on normal solid infill, otherwise it would stack up every layer and result in a bad top solid infill layer.

Here you can see some results, before and after my programm manipulated the gcode:
![IMG_0981_](https://user-images.githubusercontent.com/77171585/154022171-5da2935d-b010-4f22-b120-5536001b6901.JPG)
![IMG_0982_](https://user-images.githubusercontent.com/77171585/154022177-d1426788-709b-4394-a4dd-5a5e2f3f0b85.JPG)

Kind regards, 
Enrico aka dtex91

**Comment by supermerill** [comment]
Thin area tend to be over-extruded, i guess it comes for the infill links.
I need a bit more of data

Did you have the "enforce 100% fill volume"? (if no, can you try with it?)
Can you try with 0% infill/perimeter encroachment? (before/after)
Can you try with infill -> infill ->solid -> not connected ?

With that, i may be able to see if I can the problem at its root.
